pkg/kubectl/cmd/apply.go

```diff
@@ -245,7 +245,13 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	// Create the versioned struct from the type defined in the restmapping
 	// (which is the API version we'll be submitting the patch to)
 	versionedObject, err := api.Scheme.New(p.mapping.GroupVersionKind)
-	if err != nil {
+	switch {
+	case runtime.IsNotRegisteredError(err):
+		versionedObject, _, err = p.decoder.Decode(current, nil, nil)
```

Comment:
@liggitt Based on the[ json merge patch package](https://github.com/evanphx/json-patch) we are using, I plan to implement a generic 3-way json merge patch for patch without a schema. I guess it will be around 100 lines, if we don't count tests in.

Do you think this is reasonable?